knative-operator/pkg/controller/knativeserving/knativeserving_controller.go

```diff
@@ -130,6 +140,31 @@ func (r *ReconcileKnativeServing) configure(instance *servingv1alpha1.KnativeSer
 	return nil
 }
 
+// deleteVirtualService removes obsoleted VirtualServices.
+func (r *ReconcileKnativeServing) deleteVirtualService(instance *servingv1alpha1.KnativeServing) error {
+	labelSelector := labels.NewSelector()
+	req, err := labels.NewRequirement(routeLabelKey, selection.Exists, nil)
+	if err != nil {
+		return fmt.Errorf("failed to create requirement for label: %w", err)
+	}
+	listOpts := &client.ListOptions{LabelSelector: labelSelector.Add(*req)}
+	list := &v1alpha3.VirtualServiceList{}
+	ctx := context.TODO()
+	if err := r.client.List(ctx, listOpts, list); err != nil {
+		return err
+	}
+	for i := range list.Items {
+		if list.Items[i].GetAnnotations()[ingressClassKey] == istioIngressClass {
+			log.Info(fmt.Sprintf("deleting VirtualService %s/%s", list.Items[i].GetName(), list.Items[i].GetNamespace()))
```

Comment:
Let's just do `vs := &list.Items[i]` at the beginning of the loop and reuse that please.

Reply:
The Namespace and the Name are also in the wrong order.